patches/minecraft/net/minecraft/client/audio/SoundHandler.java.patch

```diff
@@ -0,0 +1,40 @@
+--- ../src-base/minecraft/net/minecraft/client/audio/SoundHandler.java
++++ ../src-work/minecraft/net/minecraft/client/audio/SoundHandler.java
+@@ -67,6 +67,7 @@
+     {
+         this.field_147697_e.func_148763_c();
+ 
++        Map<ResourceLocation, SoundList> resources = new java.util.IdentityHashMap<>();
```

Comment:
I would suggest using a list of `Pair` for better performance

Reply:
Or just the Map.Entry's. But this isn't going to make much of a difference, the time is dominated by IO

Reply:
@williewillus You cannot create `Map.Entry`s independently since they are dependent of maps

Reply:
You don't need to create them, they already exist.

Reply:
@williewillus meant to create and store map entries (which you cannot create) into a list. I at first suggested using some sort of pair from apache library.

Reply:
@liach `Map.Entry` is not implicitly dependent on `Map`. There are many implementations of `Map.Entry` that do not depend on being in a `Map` at all, these include 'HashMap.Node', 'ImmutableMapEntry' and all subclasses `org.apache.commons.lang3.tuple.Pair`.

Reply:
yeah. the entry set is already being iterated. you have it right [here](https://github.com/MinecraftForge/MinecraftForge/pull/4593/files#diff-c3e6ddfb5c0a348dce9ff6e24d222e15R13), literally just put that into the list/collection.